src/serac/physics/solid_mechanics.hpp

```diff
@@ -769,9 +767,68 @@ class SolidMechanics<order, dim, Parameters<parameter_space...>, std::integer_se
 
   /// @overload
   template <typename TractionType>
-  void setPiolaTraction(TractionType traction_function)
+  void setTraction(TractionType traction_function)
   {
-    setPiolaTraction(DependsOn<>{}, traction_function);
+    setTraction(DependsOn<>{}, traction_function);
+  }
+
+  /**
+   * @brief Set the pressure boundary condition
+   *
+   * @tparam PressureType The type of the pressure load
+   * @param pressure_function A function describing the pressure applied to a boundary
+   *
+   * @pre PressureType must be a object that can be called with the following arguments:
+   *    1. `tensor<T,dim> x` the reference configuration spatial coordinates for the quadrature point
+   *    2. `double t` the time (note: time will be handled differently in the future)
+   *    3. `tuple{value, derivative}`, a variadic list of tuples (each with a values and derivative),
+   *            one tuple for each of the trial spaces specified in the `DependsOn<...>` argument.
+   *
+   * @note The actual types of these arguments passed will be `double`, `tensor<double, ... >` or tuples thereof
+   *    when doing direct evaluation. When differentiating with respect to one of the inputs, its stored
+   *    values will change to `dual` numbers rather than `double`. (e.g. `tensor<double,3>` becomes `tensor<dual<...>,
+   * 3>`)
+   *
+   * @note This pressure is applied in the deformed (current) configuration if GeometricNonlinearities are on.
+   */
+  template <int... active_parameters, typename PressureType>
+  void setPressure(DependsOn<active_parameters...>, PressureType pressure_function)
+  {
+    residual_->AddBoundaryIntegral(
+        Dimension<dim - 1>{}, DependsOn<0, 1, 2, active_parameters + NUM_STATE_VARS...>{},
+        [this, pressure_function](auto X, auto displacement, auto /* acceleration */, auto shape, auto... params) {
+          // Calculate the position and normal in the shape perturbed deformed configuration
+          auto x = X + shape + 0.0 * displacement;
+
+          if (geom_nonlin_ == GeometricNonlinearities::On) {
+            x = x + displacement;
+          }
+
+          auto n = cross(get<DERIVATIVE>(x));
+
+          // serac::Functional's boundary integrals multiply the q-function output by
+          // norm(cross(dX_dxi)) at that quadrature point, but if we impose a shape displacement
+          // then that weight needs to be corrected. The new weight should be
+          // norm(cross(dX_dxi + dp_dxi)). This implies:
```

Comment:
Sorry if I'm confused here, but does p in the comment mean shape + displacement (when GeomNonlinearity is off)?

Reply:
Yes. I'll improve the comment.